packages/ra-ui-materialui/src/button/SaveButton.tsx

```diff
@@ -65,7 +65,8 @@ export const SaveButton = <RecordType extends RaRecord = any>(
     const translate = useTranslate();
     const form = useFormContext();
     const saveContext = useSaveContext();
-    const { isDirty, isValidating, isSubmitting } = useFormState();
+    const { dirtyFields, isValidating, isSubmitting } = useFormState();
+    const isDirty = Object.keys(dirtyFields).length;
```

Comment:
`useFormState().isDirty` might differ from `useFormState().dirtyFields` (https://github.com/react-hook-form/react-hook-form/issues/4740)

Reply:
Could you add that as a comment above this line? It'd help future us understand why we're not using `isDirty` here. 

also, I'm nipticking, but I'd prefer

```jsx
const isDirty = Object.keys(dirtyFields).length > 0;
```